mrmustard/physics/ansatze.py

```diff
@@ -175,6 +181,18 @@ def degree(self) -> int:
             return 0
         return self.array.shape[-1] - 1
 
+    @property
+    def polynomial_degrees(self) -> tuple[int, tuple]:
```

Comment:
A working example of your code is
`A = np.eye(2)
b = np.ones(2)
c = np.array([[[1,2,3,4,5,6],[1,2,3,4,5,0]]])

ansatze = DiffOpPolyExpAnsatz(A,b,c)
ansatze.polynomial_degrees`

But I think you meant the following, where an outer product has to be taken
`A = np.eye(2)
b = np.ones(2)
c = np.outer(np.array(([1,2,3,4,5,6])),np.array(([1,2,3,4,5,0]))).reshape(1,6,6)
ansatze = DiffOpPolyExpAnsatz(A,b,c)
ansatze.polynomial_degrees`

However, right now we don't remove trailing zeros and the degree returns one too many (starts from degree 1 rather than 0)

Reply:
Maybe it should be renamed to polynomial shape, since that is what we actually need for some other functions.